src/main/scala/fastq.scala

```diff
@@ -98,22 +132,48 @@ case object fastq {
     )
     : FASTA := ((fasta.header.type := fasta.header.Raw) :: (fasta.sequence.type := fasta.sequence.Raw) :: *[AnyDenotation]) =
       FASTA(
-        fasta.header( me.getV(id) toFastaHeader )                  ::
-        fasta.sequence( FastaSequence((me getV sequence).asString) )  :: *[AnyDenotation]
+        fasta.header( me.getV(id).toFastaHeader )                   ::
+        fasta.sequence( FastaSequence(me.getV(sequence).asString) ) :: *[AnyDenotation]
       )
 
     def toLines(implicit
-      getId: AnyApp1At[findS[AnyDenotation.Of[id.type]], RV] { type Y = id.type := id.Raw },
-      getSeq: AnyApp1At[findS[AnyDenotation.Of[sequence.type]], RV] { type Y = sequence.type := sequence.Raw },
-      getPlus: AnyApp1At[findS[AnyDenotation.Of[plus.type]],RV] { type Y = plus.type := plus.Raw },
-      getQual: AnyApp1At[findS[AnyDenotation.Of[quality.type]],RV] { type Y = quality.type := quality.Raw }
+      getId   : AnyApp1At[findS[AnyDenotation.Of[id.type]], RV] { type Y = id.type := id.Raw },
+      getSeq  : AnyApp1At[findS[AnyDenotation.Of[sequence.type]], RV] { type Y = sequence.type := sequence.Raw },
+      getPlus : AnyApp1At[findS[AnyDenotation.Of[plus.type]], RV] { type Y = plus.type := plus.Raw },
+      getQual : AnyApp1At[findS[AnyDenotation.Of[quality.type]],RV] { type Y = quality.type := quality.Raw }
     )
-    : Seq[String] =
-      Seq(
-        (me getV id)       .asString,
-        (me getV sequence) .asString,
-        (me getV plus)     .asString,
-        (me getV quality)  .asString
-      )
+    : String =
+    s"${(me getV id).asString}\n${(me getV sequence).asString}\n${(me getV plus).asString}\n${(me getV quality).asString}"
   }
+
+  def parseMapFromLines(lines: Iterator[String]): Iterator[Map[String, String]] = {
+
+    // NOTE much unsafe, should check for id and qual chars etc
+    lines.grouped(4) map {
+      quartet => {
+        Map(
+          id.label        -> quartet(0),
+          sequence.label  -> quartet(1),
+          plus.label      -> quartet(2),
+          quality.label   -> quartet(3)
+        )
```

Comment:
This is not very helpful. If you are not parsing here anything, what's the point of this method at all?